### PR TITLE
fix(issue#52): fixed Vehicle list drop down not working issue#52

### PR DESCRIPTION
**Pull request by cijokb**

### CODE REVIEW

File: src/components/ViewServiceDetails/ViewServiceDetails.js [hunk 1]
```diff
@@ -69,54 +60,55 @@ class ViewServiceDetails extends React.Component {
...
+				method: 'GET',
+				headers: {
+					'Content-Type': 'application/json'
+				}
+			}).then(function (response) {
+				return (response.text());
+			}).then(function (response) {
+				return JSON.parse(response);
+			}).then(function (response) {
+				console.log('getVehiclesList response', response.data);
```
Comment (ivinjose) [inline src/components/ViewServiceDetails/ViewServiceDetails.js]: Remove console.log

File: src/components/ViewServiceDetails/ViewServiceDetails.js [hunk 2]
```diff
@@ -69,54 +60,55 @@ class ViewServiceDetails extends React.Component {
...
+				_this.setState({
+					vehicles: response.data || []
+				});//setting empty array to prevent crash if reponse is undefined
```
Comment (ivinjose) [inline src/components/ViewServiceDetails/ViewServiceDetails.js]: The code is self explanatory here. Comments are needed only when it is complex.

File: src/components/ViewServiceDetails/ViewServiceDetails.js [hunk 3]
```diff
@@ -69,54 +60,55 @@ class ViewServiceDetails extends React.Component {
...
+				console.log('some error');
 			});
-		}).catch(function(error){
-			console.log('some error');
-		});
 	}
 
-	chooseVehicle(event, key, payload){
+	chooseVehicle(id) {
+		//let vehicleId = event.target.value;
```
Comment (ivinjose) [inline src/components/ViewServiceDetails/ViewServiceDetails.js]: Remove commented code.

File: src/components/ViewServiceDetails/ViewServiceDetails.js [hunk 4]
```diff
@@ -69,54 +60,55 @@ class ViewServiceDetails extends React.Component {
...
+				headers: {
+					'Content-Type': 'application/json'
+				}
+			}).then(function (response) {
+				return (response.text());
+			}).then(function (response) {
+				return JSON.parse(response);
+			}).then(function (response) {
+				_this.setState({
+					serviceDetails: response.data
```
Comment (ivinjose) [inline src/components/ViewServiceDetails/ViewServiceDetails.js]: Check for undefined here as well?

File: src/components/ViewServiceDetails/ViewServiceDetails.js [hunk 5]
```diff
@@ -22,43 +22,34 @@ class ViewServiceDetails extends React.Component {
...
-							onChange={this.chooseVehicle.bind(this)}
-							renderValue={selected => {
-								// debugger;
-								if (selected && selected.length === 0) {
-									return <em>Placeholder</em>;
-								}
-								return selected.join(', ');
-							}}
+						<Select
+							displayEmpty
```
Comment (ivinjose) [inline src/components/ViewServiceDetails/ViewServiceDetails.js]: Understand what displayEmpty is for and then use accordingly.

File: src/components/ViewServiceDetails/ViewServiceDetails.js [hunk 6]
```diff
@@ -22,43 +22,34 @@ class ViewServiceDetails extends React.Component {
...
-		if( queryParams && queryParams.vehicle ){
-			this.chooseVehicle(null,null, queryParams.vehicle);
+		if (queryParams && queryParams.vehicle) {
+			this.chooseVehicle(queryParams.vehicle);
 		}
 	}
 
 	render() {
+		let placeHolderItem = <MenuItem disabled value="" key='chooseVehicle'><em>Choose Vehicle</em></MenuItem>;
+		const menuItems = [placeHolderItem, ...this.state.vehicles.map((vehicle) => (
```
Comment (ivinjose) [inline src/components/ViewServiceDetails/ViewServiceDetails.js]: Good code. 👍

File: src/components/ViewServiceDetails/ViewServiceDetails.css [hunk 7]
```diff
@@ -2,4 +2,8 @@
     .body{
         margin: 16px;
     }
+}
+
+.select-cmp {
+    width:100%;
```
Comment (ivinjose) [inline src/components/ViewServiceDetails/ViewServiceDetails.css]: Watch out for spacing between width: and 100%

File: src/components/AddVehicleDetails/AddVehicleDetails.js [hunk 8]
```diff
@@ -71,15 +69,13 @@ class AddVehicleDetails extends React.Component {
...
 
 	saveVehicle(e){
-		let { store } = this.context;
-		let vehicle = { name: this.state.vehicle };
+		let data = [{ name: this.state.vehicle }];
 
-		let data = Object.assign( {}, vehicle );
 		let _this = this;
 
 		console.log('data',data);
```
Comment (ivinjose) [inline src/components/AddVehicleDetails/AddVehicleDetails.js]: Remove console.log

File: src/components/AddVehicleDetails/AddVehicleDetails.js [hunk 9]
```diff
@@ -55,12 +55,10 @@ class AddVehicleDetails extends React.Component {
 		);
 	}
 
-	updateVehicle(event, newValue){
+	updateVehicle(event){
 		//TODO:: Debug why the data is not getting saved into DB.
```
Comment (ivinjose) [inline src/components/AddVehicleDetails/AddVehicleDetails.js]: Remove console.log

File: src/components/ViewServiceDetails/ViewServiceDetails.js (hunk 1 above)
Comment (cijokb) [reply src/components/ViewServiceDetails/ViewServiceDetails.js]: done

File: src/components/ViewServiceDetails/ViewServiceDetails.js (hunk 2 above)
Comment (cijokb) [reply src/components/ViewServiceDetails/ViewServiceDetails.js]: removed the comments

File: src/components/ViewServiceDetails/ViewServiceDetails.js (hunk 3 above)
Comment (cijokb) [reply src/components/ViewServiceDetails/ViewServiceDetails.js]: removed

File: src/components/ViewServiceDetails/ViewServiceDetails.js (hunk 4 above)
Comment (cijokb) [reply src/components/ViewServiceDetails/ViewServiceDetails.js]: Added check for undefined

File: src/components/ViewServiceDetails/ViewServiceDetails.js (hunk 5 above)
Comment (cijokb) [reply src/components/ViewServiceDetails/ViewServiceDetails.js]: it allows displaying default value (currently it is displaying 'choose vehicle'), if nothing is selected

File: src/components/ViewServiceDetails/ViewServiceDetails.js (hunk 6 above)
Comment (cijokb) [reply src/components/ViewServiceDetails/ViewServiceDetails.js]: Thanks

File: src/components/ViewServiceDetails/ViewServiceDetails.css (hunk 7 above)
Comment (cijokb) [reply src/components/ViewServiceDetails/ViewServiceDetails.css]: added the format

File: src/components/AddVehicleDetails/AddVehicleDetails.js (hunk 8 above)
Comment (cijokb) [reply src/components/AddVehicleDetails/AddVehicleDetails.js]: removed

File: src/components/AddVehicleDetails/AddVehicleDetails.js (hunk 9 above)
Comment (cijokb) [reply src/components/AddVehicleDetails/AddVehicleDetails.js]: removed

**Pull request merged**